### ISSUE TITLE
Implement Batch Renormalization in TF

### ISSUE THREAD

**Issue by cancan101** [tensorflow/tensorflow#7476]
See paper: https://arxiv.org/pdf/1702.03275.pdf
Keras implementation: https://github.com/ajbrock/Neural-Photo-Editor/blob/master/layers.py#L35

**Comment by cancan101** [comment]
Looks like it has been implemented in tensorpack: https://github.com/ppwwyyxx/tensorpack/blob/3f238a015b941041e58ed43e63d5306dbae979fc/tensorpack/models/batch_norm.py#L208

**Comment by hengck23** [comment]
how can we set the values of rmax and dmax?

**Comment by shiyemin** [comment]
@cancan101 @hengck23 [Here](https://github.com/shiyemin/shuttleNet/blob/master/ops/ops.py#L60-L316) is my implementation. I also provide my code for set the RMAX and DMAX.

Basic usage:
`batch_renorm_decay=0.9
batch_renorm_epsilon=0.001

global_step = tf.train.get_global_step()
RMAX_decay = ops.adjust_max(global_step, 200, 800, 1, 3, name='RMAXDECAY')
DMAX_decay = ops.adjust_max(global_step, 200, 600, 1, 5, name='DMAXDECAY')
batch_renorm_params = {
    'renorm': True,
    'RMAX': RMAX_decay,
    'DMAX': DMAX_decay,
    'decay': batch_renorm_decay,
    'epsilon': batch_renorm_epsilon,
}
with slim.arg_scope([slim.conv2d, slim.fully_connected], activation_fn=tf.nn.relu,
                    normalizer_fn=ops.fused_batch_norm,
                    normalizer_params=batch_renorm_params) as scope:
`

BTW, my implementation is fully compatible with TF's batch_norm.

**Comment by shkr** [comment]
I have submitted a quick PR with the renorm parameter

**Comment by taion** [comment]
Looks like this got added in https://github.com/tensorflow/tensorflow/pull/9059/commits/55c809b62d688637813c828301e528cb2fbcc54b.

**Comment by ygoncharov** [comment]
But no support for fused batch norm :(

**Comment by cancan101** [comment]
This issue solved. Any future discussion re: fused batch re-norm can be found at: https://github.com/tensorflow/tensorflow/issues/9141.
